### PR TITLE
Allow deleting of archived triggers

### LINKED ISSUE #4520: Allow deleting of ALL archived triggers
- add a "delete" bulk action item
- add a "delete all" content menu item

### LINKED ISSUE #4550: Allow deleting of all archived triggers
add a "delete all" content menu item - should allow users to delete ALL archived triggers all at once

### PR DESCRIPTION
**Pull request by susanm74**
fixes #4520 
fixes #4550

### CODE REVIEW

File: temba/triggers/models.py [hunk 1]
```diff
@@ -351,6 +351,11 @@ def apply_action_restore(cls, user, triggers):
                 trigger.restore(user)
                 trigger_scopes = trigger_scopes | trigger_scope
 
+    @classmethod
+    def apply_action_delete(cls, user, triggers):
+        for trigger in triggers:
+            trigger.delete()
```
Comment (norkans7) [inline temba/triggers/models.py:357]: I think it is best we make sure we never delete a trigger that not archived here

Comment (norkans7) [reply temba/triggers/models.py:357]: never mind those will have been filtered out by the queryset, so that check is not needed here

Comment (norkans7) [reply temba/triggers/models.py:357]: but we need a test for that

File: temba/triggers/tests.py [hunk 2]
```diff
@@ -1526,6 +1527,61 @@ def test_archived(self):
...
+            self.admin,
+            Trigger.TYPE_KEYWORD,
+            flow,
+            keyword="ringo",
+            match_type=Trigger.MATCH_ONLY_WORD,
+            is_archived=True,
+        )
+
+        # can bulk delete archived triggers
+        self.client.post(archived_url, {"action": "delete", "objects": [trigger3.id, trigger4.id]})
```
Comment (norkans7) [inline temba/triggers/tests.py:1569]: Add a trigger that is not archived above and try to pass its ID here and assert that it should not be deleted even for the case for delete all

File: temba/triggers/models.py (hunk 1 above)
Comment (susanm74) [reply temba/triggers/models.py:357]: @norkans7 do you mean like adding an `assert trigger.is_archived, "can only delete archived triggers" inside the for loop? similar to this line https://github.com/nyaruka/rapidpro/blob/e3c87598e9b1d4198bae1bbcbf74ef161f693cd4/temba/triggers/views.py#L465 in this closed PR? https://github.com/nyaruka/rapidpro/pull/4517

**Review by norkans7 — COMMENTED** [review]
Looks good, just one more unit test, that is nice to have

File: temba/triggers/models.py (hunk 1 above)
Comment (norkans7) [reply temba/triggers/models.py:357]: No, that is not needed we can just have a unit test for that, the get_queryset method will make sure the triggers not archived are excluded by the time we call `apply_action_delete`

File: temba/triggers/tests.py [hunk 3]
```diff
@@ -1526,6 +1528,83 @@ def test_archived(self):
...
+            self.admin,
+            Trigger.TYPE_KEYWORD,
+            flow,
+            keyword="simon",
+            match_type=Trigger.MATCH_ONLY_WORD,
+            is_active=True,
+        )
+
+        # can bulk delete archived triggers
+        self.client.post(archived_url, {"action": "delete", "objects": [trigger3.id, trigger4.id]})
```
Comment (norkans7) [inline temba/triggers/tests.py:1580]: we need the test to pass the `trigger7.id` too in this list of objects

Comment (norkans7) [reply temba/triggers/tests.py:1580]: so mixing archived IDs and active IDs so to make sure active IDs are excluded

Comment (susanm74) [reply temba/triggers/tests.py:1580]: @norkans7 i actually pass it a bit further down on line 1589

Comment (susanm74) [reply temba/triggers/tests.py:1580]: @norkans7 but i can make that a bit more concise based on your comment

**Review by norkans7 — APPROVED** [review]

File: temba/triggers/views.py [hunk 4]
```diff
@@ -524,15 +524,23 @@ def pre_process(self, request, *args, **kwargs):
...
-        bulk_actions = ("restore",)
+        bulk_actions = ("restore", "delete")
         title = _("Archived Triggers")
         menu_path = "/trigger/archived"
 
+        def build_content_menu(self, menu):
+            menu.add_js(
+                "triggers_delete_all",
+                _("Delete All"),
+                "confirmDeleteAllArchivedTriggers(event)",
```
Comment (ericnewcomer) [inline temba/triggers/views.py:540]: We want to get rid of javascript method names in content menu items

File: templates/triggers/trigger_list.haml [hunk 5]
```diff
@@ -81,6 +98,14 @@
   {{ block.super }}
 
   :javascript
+    var menu = document.querySelector("temba-content-menu");      
+    menu.addEventListener("temba-selection", function(event) {
+      var item = event.detail;
+      if (item.id == "triggers_delete_all") {
+        confirmDeleteAllArchivedTriggers();
+      }
+    });
```
Comment (ericnewcomer) [inline templates/triggers/trigger_list.haml:107]: This looks correct and should be all you really need

**Review by ericnewcomer — APPROVED** [review]

**Comment by ericnewcomer** [comment]
@rowanseymour this is ready for merge, @susanm74 will address my comments in a separate PR

**Comment by susanm74** [comment]
@ericnewcomer @rowanseymour https://github.com/nyaruka/rapidpro/issues/4568

**Pull request merged**